stencil-library/src/components/table/table.tsx

```diff
@@ -0,0 +1,138 @@
+import { Component, Host, h, Prop } from '@stencil/core';
+import { PagingInfo, pagingDefaults } from '../table/table-utils';
+
+interface ExtendedPagingInfo extends PagingInfo {
+  onPrev: () => void;
+  onNext: () => void;
+}
+
+const ExtendedPagingDefaults = {
+  ...pagingDefaults,
+  onPrev: () => {},
+  onNext: () => {},
+}
+
+export interface TableProps {
+  loading: boolean;
+  errorMessage: string;
+  columnData: string[];
+  rowData: any[];
+  paging: ExtendedPagingInfo;
+}
+
+@Component({
+  tag: 'justifi-table',
+  styleUrl: './table.scss',
+  shadow: true,
+})
+
+export class Table {
+  @Prop() loading: TableProps['loading'] = true;
+  @Prop() errorMessage: TableProps['errorMessage'] = '';
+  @Prop() rowData: TableProps['rowData'] = [];
+  @Prop() columnData: TableProps['columnData'] = [];
+  @Prop() paging: TableProps['paging'] = ExtendedPagingDefaults;
+
+  showEmptyState() {
+    return this.rowData ? this.rowData.length < 1 : true;
+  }
+
+  emptyState = (
+    <tr>
+      <td class="empty-state" part="empty-state" colSpan={this.columnData.length} style={{ textAlign: 'center' }}>No payments to show</td>
+    </tr>
+  );
+
+  errorState = () => (
+    <tr>
+      <td class="error-state" part="error-state" colSpan={this.columnData.length} style={{ textAlign: 'center' }}>
+        An unexpected error occurred: {this.errorMessage}
+      </td>
+    </tr>
+  );
+
+  loadingState = (
+    <tr>
+      <td class="loading-state" part="loading-state-cell" colSpan={this.columnData.length} style={{ textAlign: 'center' }}>
+        <div part="loading-state-spinner" class="spinner-border" role="status">
+          <span class="visually-hidden">Loading...</span>
+        </div>
+      </td>
+    </tr>
+  );
+
+  paginationBar = () => {
+    return (
+      <div class="pagination-bar d-flex justify-content-center gap-3">
```

Comment:
Are `pagination-bar`, `pagination-btn`, `pagination-prev-btn` etc being used? I don't see the styles anywhere

Reply:
Nope, can be removed safely.